packages/opentelemetry-web/src/WebTracer.ts

```diff
@@ -14,25 +14,43 @@
  * limitations under the License.
  */
 
+import { BasePlugin } from '@opentelemetry/core';
 import { BasicTracer, BasicTracerConfig } from '@opentelemetry/tracing';
 import { StackScopeManager } from './StackScopeManager';
 
+/**
+ * WebTracerConfig provides an interface for configuring a Web Tracer.
+ */
+export interface WebTracerConfig extends BasicTracerConfig {
+  /**
+   * plugins to be used with tracer, they will be enabled automatically
+   */
+  plugins?: BasePlugin<unknown>[];
+}
+
 /**
  * This class represents a web tracer with {@link StackScopeManager}
  */
 export class WebTracer extends BasicTracer {
   /**
    * Constructs a new Tracer instance.
+   * @param config Web Tracer config
    */
-  /**
-   *
-   * @param {BasicTracerConfig} config Web Tracer config
-   */
-  constructor(config: BasicTracerConfig = {}) {
+  constructor(config: WebTracerConfig = {}) {
     if (typeof config.scopeManager === 'undefined') {
       config.scopeManager = new StackScopeManager();
     }
-    config.scopeManager.enable();
+    if (typeof config.plugins === 'undefined') {
+      config.plugins = [];
+    }
     super(Object.assign({}, { scopeManager: config.scopeManager }, config));
+
+    // enable scope manager
```

Comment:
Optional: what would you think about removing this comment and the `// enable all plugins` comment below? I tend to recommend line comments like this when the code is confusing in some way, but in these cases, the code seems pretty clear to me.